django/tasks/task.py

```diff
@@ -0,0 +1,271 @@
+from dataclasses import dataclass, field, replace
+from datetime import datetime
+from inspect import iscoroutinefunction
+from typing import Any, Callable, Dict, Optional, Type
+
+from asgiref.sync import async_to_sync, sync_to_async
+
+from django.db.models.enums import TextChoices
+from django.utils.translation import gettext_lazy as _
+
+from .exceptions import ResultDoesNotExist
+from .utils import get_module_path, json_normalize
+
+DEFAULT_TASK_BACKEND_ALIAS = "default"
+DEFAULT_QUEUE_NAME = "default"
+MIN_PRIORITY = -100
+MAX_PRIORITY = 100
+DEFAULT_PRIORITY = 0
+
+TASK_REFRESH_ATTRS = {
+    "_exception_class",
+    "_traceback",
+    "_return_value",
+    "finished_at",
+    "started_at",
+    "status",
+    "enqueued_at",
+}
+
+
+class ResultStatus(TextChoices):
+    NEW = ("NEW", _("New"))
```

Comment:
Don't know what Django policy is, but for such small strings I find it helpful to use `pgettext_lazy` instead of `gettext_lazy`, to give translators a hint what they're translating, e.g. some languages have gender and such.

Reply:
I've not worked with translations too much in Django. @sarahboyce can you advise? :)

Reply:
We don't have a policy. `pgettext_lazy` allows you to add additional context. I don't have a suggestion as to what context you should add so I would leave this